mlir/utils/performance/analysis/compileAndCollectTuningData.py

```diff
@@ -0,0 +1,795 @@
+            converted_args.extend(["-t", dataType])
+            i += 1
+            continue
+
+        if i + 1 < len(test_args):
+            opt = test_args[i]
+            val = test_args[i + 1]
+            
+            # Map short form arguments to rocmlir-gen long form
+            if opt == "-n":
+                converted_args.extend(["--batchsize", val])
+            elif opt == "-c":
+                converted_args.extend(["--in_channels", val])
+            elif opt == "-H":
+                converted_args.extend(["--in_h", val])
+            elif opt == "-W":
+                converted_args.extend(["--in_w", val])
+            elif opt == "-k":
+                converted_args.extend(["--out_channels", val])
+            elif opt == "-y":
+                converted_args.extend(["--fil_h", val])
+            elif opt == "-x":
+                converted_args.extend(["--fil_w", val])
+            elif opt == "-p":
+                converted_args.extend(["--padding_h", val])
+            elif opt == "-q":
+                converted_args.extend(["--padding_w", val])
+            elif opt == "-u":
+                converted_args.extend(["--conv_stride_h", val])
+            elif opt == "-v":
+                converted_args.extend(["--conv_stride_w", val])
+            elif opt == "-l":
+                converted_args.extend(["--dilation_h", val])
+            elif opt == "-j":
+                converted_args.extend(["--dilation_w", val])
+            elif opt == "-g":
+                converted_args.extend(["-g", val])
+            elif opt == "-f":
+                converted_args.extend(["--fil_layout", val.lower()])
+            elif opt == "-I":
+                converted_args.extend(["--in_layout", val.lower()])
+            elif opt == "-O":
+                converted_args.extend(["--out_layout", val.lower()])
+            elif opt == "-F":
+                # Convert direction flag to operation
+                direction_val = int(val)
+                if direction_val == 1:
+                    operation = "conv"
+                elif direction_val == 2:
+                    operation = "conv_bwd_data"
+                elif direction_val == 4:
+                    operation = "conv_bwd_weight"
+            else:
+                # Unknown argument, do not add
+                pass
+            i += 2
+    
+    return [converted_args, operation]
+
+def compile_config(config, operation, binaries, timestamp):
+    arch = config["# arch"].split(':')[0]
+    num_cu = config["numCUs"]
+
+    # Parse and add the test vector arguments
+    test_vector = config["testVector"]
+    test_args = parse_test_args(test_vector)
+
+    # If operation is a convolution, we need to convert the test_args to a
+    # format that rocmlir-gen can understand
+    if operation.lower() == 'conv':
+        [test_args, operation] = convertConvTestArgs(test_args, operation)
+
+    # Build the rocmlir-gen command
+    rocmlir_gen_cmd = [
+        binaries[0],
```

Comment:
keep binaries a dict/data class ? I can't tell if `binaries[0]` is rocmlir-gen or rocmlir-driver . 

Or use perfRunner 
https://github.com/ROCm/rocMLIR/blob/f9986dc487d70ae223efefdb24b7955353fec9f9/mlir/utils/performance/perfRunner.py#L68